x/interchainstaking/keeper/withdrawal_record.go

```diff
@@ -52,6 +52,9 @@ func (k *Keeper) GetNextWithdrawalRecordSequence(ctx sdk.Context) uint64 {
 
 func (k *Keeper) AddWithdrawalRecord(ctx sdk.Context, chainID, delegator string, distributions []*types.Distribution, recipient string, burnAmount sdk.Coin, hash string, status int32, completionTime time.Time, epochNumber int64) {
 	record := types.WithdrawalRecord{ChainId: chainID, Delegator: delegator, Distribution: distributions, Recipient: recipient, Status: status, BurnAmount: burnAmount, Txhash: hash, CompletionTime: completionTime, EpochNumber: epochNumber}
+	if record.BurnAmount.IsNegative() || record.BurnAmount.IsZero() {
+		panic(fmt.Errorf("burnAmount cannot be negative or zero"))
```

Comment:
Panicking is generally not a good idea; we should change the signature here to return an error in the event that BurnAmount is not positive.